packages/create-cloudflare/src/helpers/command.ts

```diff
@@ -254,18 +263,34 @@ export const npmInstall = async () => {
 	});
 };
 
+const needsReset = (ctx: PagesGeneratorContext) => {
+	const { npm } = detectPackageManager();
+	const projectPath = ctx.project.path;
+
+	switch (npm) {
+		case "npm":
+			return false;
+		case "yarn":
+			return !existsSync(path.join(projectPath, "yarn.lock"));
+		case "pnpm":
+			return !existsSync(path.join(projectPath, "pnpm-lock.yaml"));
+		case "bun":
+			return !existsSync(path.join(projectPath, "bun.lockb"));
+	}
+};
+
 // Resets the package manager context for a project by clearing out existing dependencies
 // and lock files then re-installing.
 export const resetPackageManager = async (ctx: PagesGeneratorContext) => {
 	const { npm } = detectPackageManager();
 
-	// Only do this when using pnpm or yarn
-	if (npm === "npm") return;
+	if (!needsReset(ctx)) return;
```

Comment:
Surely we have a lint rule that stops if blocks not in curlies?
I know these are only single line ifs, and I'm being a bit pedantic, but this is a very common cause of errors.

Reply:
I agree it's does feel a bit unsafe (although I understand the convenience)

I don't think we have the [eslint rule](https://eslint.org/docs/latest/rules/curly) set and actually we do this sort of thing in a bunch of different places (C3 not being the biggest offender):
![Screenshot 2023-11-02 at 12 49 19](https://github.com/cloudflare/workers-sdk/assets/61631103/cb1937fb-5c67-4a26-a536-786707a3db52)

I'm totally on board adding the rule in a followup PR if devprod's on board with it 🙂